client/server/silicon/silicon.ts

```diff
@@ -0,0 +1,68 @@
+import { ApolloQueryResult, DocumentNode } from '@apollo/client/core';
+import { getOperationName } from '@apollo/client/utilities';
+import { apolloClient } from '../utils/subscription_manager';
+import { MarketModelsResponse, EncryptedMarketModelsResponse, Market, EncryptedMarket } from "../../graphql/graphql";
+
+
+export class Silicon<T> {
+    protected query: any;
+    public markets: T[];
+
+    constructor(query: any) {
+        this.query = query;
+        this.markets = [];
+    }
+
+    async fetchMarkets(): Promise<void> {
+        throw new Error("Method 'fetchMarkets' must be implemented.");
+    }
+}
+
+class TransparentSilicon extends Silicon<Market> {
+    async fetchMarkets() {
```

Comment:
please add natural language comments on top of every function (if you're keeping these)

Reply:
otherwise, can label as 
```
/*
 * [TMP]
 */
```

Reply:
still missing all the comments :)